### PR TITLE
Merge/upstream

### PR DESCRIPTION
**Pull request by jordy25519**
Latest round of upstream changes.

Adds:
- `srml/nicks` provides a simple nick-naming and lookup service for `AccountId`s. Not intended as an ENS implementation
- Contract "get storage" RPC endpoints for querying smart contract storage values: https://github.com/plugblockchain/plug-blockchain/commit/155c217178cc8e3cf4da668dd54e28a8b113eb6b
- SECP256k1/ECDSA singing for extrinsics: https://github.com/paritytech/substrate/pull/3861
- `srml/evm` module to support and exec EVM contracts
- Faster binary builds if git HEAD has not changed

Changes:
- Updates `impl-serde` version
- Transaction version bumped to `4`
- `im_online` heartbeat message priority increased
- `AnySignature` renamed to `MultiSignature`
- Other misc. bug fixes and refactors

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -63,6 +63,7 @@ jobs:
   build:
     machine:
       image: ubuntu-1604:201903-01
+    resource_class: large
```
Comment (cowboy-bebug) [inline .circleci/config.yml:66]: 🎉

File: Cargo.toml [hunk 2]
```diff
@@ -1,22 +1,3 @@
...
-edition = "2018"
-
-[dependencies]
-cli = { package = "node-cli", path = "node/cli" }
-futures = "0.1.29"
-ctrlc = { version = "3.1.3", features = ["termination"] }
-
-[build-dependencies]
-vergen = "3.0.4"
-
```
Comment (cowboy-bebug) [inline Cargo.toml:19]: Do we want to keep these lines even though upstream deleted them?

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,528 @@
+= Substrate
```
Comment (cowboy-bebug) [inline README.adoc]: I think we keep this file removed 😄

**Comment by jordy25519** [comment]
So the failing test is caused by slow block production because tests are compiled in debug mode 

```rust
[2019-11-06T03:39:37Z DEBUG substrate_consensus_slots] Starting authorship at slot 1573011576; timestamp = 1573011576030
[2019-11-06T03:39:38Z INFO  substrate_consensus_slots] Discarding proposal for slot 1573011576; block production took too long
[2019-11-06T03:39:38Z INFO  substrate_consensus_slots] Recompile your node in `--release` mode to mitigate this problem.
[2019-11-06T03:39:40Z DEBUG substrate_consensus_slots] Starting authorship at slot 1573011579; timestamp = 1573011579624
[2019-11-06T03:39:43Z INFO  substrate_consensus_slots] Discarding proposal for slot 1573011579; block production took too long
[2019-11-06T03:39:43Z INFO  substrate_consensus_slots] Recompile your node in `--release` mode to mitigate this problem.
[2019-11-06T03:39:47Z DEBUG substrate_consensus_slots] Starting authorship at slot 1573011585; timestamp = 1573011585904
[2019-11-06T03:39:50Z INFO  substrate_consensus_slots] Discarding proposal for slot 1573011585; block production took too long
[2019-11-06T03:39:50Z INFO  substrate_consensus_slots] Recompile your node in `--release` mode to mitigate this problem.
```

I'm thinking it's best to `#[ignore]` this test for now rather than change the whole test step to use release builds

**Review by cowboy-bebug — APPROVED** [review]
Yep looks good! I have a couple of comments

File: Cargo.toml (hunk 2 above)
Comment (jordy25519) [reply Cargo.toml:19]: Yea so upstream moved all of this stuff into `node/cli/Cargo.toml`

**Pull request merged**